framework/validators/FileValidator.php

```diff
@@ -412,7 +413,11 @@ protected function validateExtension($file)
             }
         }
 
-        if (!in_array($extension, $this->extensions, true)) {
+        if (is_array($this->extensions) and count($this->extensions)) {
+            foreach ($this->extensions as $ext) {
+                if (StringHelper::endsWith($file->name, $ext, false))
+                    return true;
+            }
```

Comment:
```suggestion
        if (!empty($this->extensions)) {
            foreach ((array) $this->extensions as $ext) {
                if (StringHelper::endsWith($file->name, ".$ext", false))
                    return true;
            }
```

Reply:
This actually could be simplified to:

```php
        foreach ($this->extensions as $ext) {
            if (StringHelper::endsWith($file->name, ".$ext", false)) {
                return true;
            }
        }
```
since `$extensions` is normalized to array in `init()`.